### PR TITLE
test(global-header): playwright refactor

### PR DESCRIPTION
**Pull request by ZhuoyuJin**
### Proposed behaviour

Refactor Global Header tests from Cypress to Playwright

### Current behaviour

Tests are currently Cypress

### Checklist

- [x] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [ ] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [x] Playwright automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required
- [ ] Related docs have been updated if required

#### QA

- [ ] Tested in CodeSandbox/storybook
- [ ] Add new Cypress test coverage if required
- [x] Carbon implementation matches Design System/designs
- [x] UI Tests GitHub check reviewed if required

### Additional context

N/A

### Testing instructions
- [x] Run Global Header tests in Playwright Test Runner to check if the `global-header.pw.tsx` file passed
- [x] Run all other tests in Playwright Test Runner to check none of the other `*.pw.tsx` files have regressed

### CODE REVIEW

File: src/components/global-header/global-header.pw.tsx [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    const NavIndex = parseInt(navigationBarZIndex.toString());
+
+    expect(globalIndex).toBeGreaterThan(NavIndex);
+  });
+
+  test("should check when logo prop is passed, the height of the logo element never exceeds the maximum height of the component", async ({
+    mount,
+    page,
+  }) => {
+    const expectedHeight = 40;
```
Comment (DipperTheDan) [inline src/components/global-header/global-header.pw.tsx:56]: **suggestion(non-blocking):** As this only used in the assertion on line 62, you could just remove this and have the value there? If this goes against your agree styling/convention that the QA's have agreed, I'm happy as it is 😃

File: src/components/global-header/global-header.pw.tsx [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+    await mount(<GlobalHeaderWithLogo />);
+
+    await expect(globalHeaderLogo(page)).toHaveCSS(
+      "height",
+      `${expectedHeight}px`
+    );
```
Comment (DipperTheDan) [inline src/components/global-header/global-header.pw.tsx:63]: So if you implement my previous comment, this should work. 

```suggestion
    await expect(globalHeaderLogo(page)).toHaveCSS(
      "height",
      `40px`
    );
```

**Review by DipperTheDan — DISMISSED** [review]

File: src/components/global-header/global-header.pw.tsx [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+} from "./component.test-pw";
+import navigationBar from "../../../playwright/components/navigation-bar";
+import {
+  globalHeader,
+  globalHeaderLogo,
+} from "../../../playwright/components/global-header";
+import { checkAccessibility } from "../../../playwright/support/helper";
+
+test.describe("Global Header component", () => {
+  test("should not cause a ResizeObserver-related error to occur", async ({
```
Comment (stephenogorman) [inline src/components/global-header/global-header.pw.tsx:16]: `test("should render without causing a ResiveObserver related error"`

File: src/components/global-header/global-header.pw.tsx [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+  }) => {
+    await mount(<GlobalHeaderWithErrorHandler />);
+
+    await page.waitForTimeout(500);
+
+    const errorState = page.locator("#error-div");
+    await expect(errorState).toHaveText("");
+  });
+
+  test("should check that z-index of component is greater than that of NavigationBar", async ({
```
Comment (stephenogorman) [inline src/components/global-header/global-header.pw.tsx:28]: `test("should render with z-index of component greater than z-index of NavigationBar"`

File: src/components/global-header/global-header.pw.tsx [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+      }
+    );
+
+    const globalIndex = parseInt(globalHeaderZIndex.toString());
+    const NavIndex = parseInt(navigationBarZIndex.toString());
+
+    expect(globalIndex).toBeGreaterThan(NavIndex);
+  });
+
+  test("should check when logo prop is passed, the height of the logo element never exceeds the maximum height of the component", async ({
```
Comment (stephenogorman) [inline src/components/global-header/global-header.pw.tsx:52]: `test("should render with height of logo element never exceeding the max height of the component when logo prop is passed"`

File: src/components/global-header/global-header.pw.tsx [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+    mount,
+    page,
+  }) => {
+    await mount(<GlobalHeaderWithLogo />);
+
+    await expect(globalHeaderLogo(page)).toHaveCSS("height", `40px`);
+  });
+
+  test.describe("Accessibility tests", () => {
+    test("should pass accessibility tests for Global-Header FullMenuExample", async ({
```
Comment (stephenogorman) [inline src/components/global-header/global-header.pw.tsx:62]: ` test("should pass tests for FullMenuExample"`

File: src/components/global-header/global-header.pw.tsx [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+      mount,
+      page,
+    }) => {
+      await mount(<FullMenuExample />);
+
+      await checkAccessibility(page);
+    });
+
+    test("should pass accessibility tests for Global-Header when logo prop is passed", async ({
```
Comment (stephenogorman) [inline src/components/global-header/global-header.pw.tsx:71]: `test("should pass tests for GlobalHeaderWithLogo example"`

**Review by stephenogorman — APPROVED** [review]

**Review by DipperTheDan — APPROVED** [review]

**Pull request merged**